src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -122,13 +122,14 @@ public String getScript() {
         if(os == OsType.ZOS) {
             Charset zOSSystemEncodingCharset = Charset.forName(ws.act(new getIBMzOsEncoding()));
             if(SYSTEM_DEFAULT_CHARSET.equals(getCharset())) {
-            // Setting default charset to IBM z/OS default EBCDIC charset on z/OS if no encoding specified on sh step
-            charset(zOSSystemEncodingCharset);
+                // Setting default charset to IBM z/OS default EBCDIC charset on z/OS if no encoding specified on sh step
+                charset(zOSSystemEncodingCharset);
             }
-            scriptEncodingCharset = zOSSystemEncodingCharset != null ? zOSSystemEncodingCharset.name() : scriptEncodingCharset;
+            scriptEncodingCharset = zOSSystemEncodingCharset.name();
         }
-
+        
         ShellController c = new ShellController(ws);
+        c.isZos = (os == OsType.ZOS);
```

Comment:
For stylistic purposes, nicer to make the field `final` and move its initialization to the constructor.

Reply:
done